contentgrid-spring-data-rest/src/main/java/com/contentgrid/spring/data/rest/problem/ContentGridExceptionHandler.java

```diff
@@ -101,6 +100,26 @@ ResponseEntity<Problem> handleRepositoryConstraintViolationException(RepositoryC
     }
 
     @ExceptionHandler
+    ResponseEntity<Problem> handleHttpMessageReadException(@NonNull HttpMessageNotReadableException exception) {
+        Throwable currentException = exception;
+
+        while (currentException != null) {
+            if (currentException instanceof JsonMappingException mappingException) {
+                return handleMappingException(mappingException);
+            } else if (currentException instanceof JsonParseException parseException) {
+                return handleJsonParseException(parseException);
+            }
+            currentException = currentException.getCause();
+        }
+
+        // Fallback handler: just a generic bad request
+        return responseEntityFactory.createResponse(
+                problemFactory.createProblem(ProblemType.INVALID_REQUEST_BODY)
+                        .withStatus(HttpStatus.BAD_REQUEST)
+                        .withDetail(exception.getMessage())
```

Comment:
Sonar complains about `exception` being null here, but it's not `currentException`, which could actually be null.